crates/sui-storage/src/event_store/sql.rs

```diff
@@ -168,131 +187,186 @@ impl SqlEventStore {
-    match raw_bytes {
-        Some(bytes) => {
-            let num_bytes = bytes.len();
-            Ok(Some(ObjectID::try_from(bytes).map_err(|_e| {
-                SuiError::BadObjectType {
-                    error: format!("Could not parse {} bytes into ObjectID", num_bytes),
-                }
-            })?))
+    fn try_extract_object_id(row: &SqliteRow, col: usize) -> Result<Option<ObjectID>, SuiError> {
+        let raw_bytes: Option<Vec<u8>> = row.get(col);
+        match raw_bytes {
+            Some(bytes) => Ok(Some(ObjectID::try_from(bytes.as_slice()).map_err(
+                |_e| SuiError::BadObjectType {
+                    error: format!("Could not parse bytes {:?} into ObjectID", bytes),
+                },
+            )?)),
+            None => Ok(None),
+        }
+    }
+
+    fn try_extract_sender_address(row: &SqliteRow) -> Result<Option<SuiAddress>, SuiError> {
+        let raw_bytes: Option<Vec<u8>> = row.get(EventsTableColumns::Sender as usize);
+        match raw_bytes {
+            Some(bytes) => Ok(Some(SuiAddress::try_from(bytes.as_slice()).map_err(
+                |_e| SuiError::BadObjectType {
+                    error: format!(
+                        "Could not parse sender address bytes: {:?} into SuiAddress",
+                        bytes
+                    ),
+                },
+            )?)),
+            None => Ok(None),
+        }
+    }
+
+    fn try_extract_recipient(row: &SqliteRow) -> Result<Option<Owner>, SuiError> {
+        let deserialized_owner: Option<String> = row.get(EventsTableColumns::Recipient as usize);
+
+        match deserialized_owner {
+            Some(owner_string) => Ok(Some(serde_json::from_str::<Owner>(&owner_string).map_err(
+                |_e| SuiError::BadObjectType {
+                    error: format!(
+                        "Could not deserialize owner str: {:?} into Owner",
+                        owner_string
+                    ),
+                },
+            )?)),
+            None => Ok(None),
+        }
+    }
+
+    // convert an event's extra fields into a stringifed JSON Value.
+    fn event_to_json(event: &EventEnvelope) -> String {
+        // For move events, we only store the move_struct_json_value
+        if let Some(json_value) = &event.move_struct_json_value {
+            json_value.to_string()
+        } else {
+            // For non-move-events, extract whatever we can to rebuild the event
+            // and store them
+            let mut fields = BTreeMap::new();
+            if let Some(transfer_type_u16) = event
+                .event
+                .transfer_type()
+                .map(|tt| TransferTypeVariants::from(tt) as u16)
+            {
+                fields.insert(TRANSFER_TYPE_KEY, transfer_type_u16.to_string());
+            };
+            if let Some(object_version) = event.event.object_version().map(|ov| ov.value()) {
+                fields.insert(OBJECT_VERSION_KEY, object_version.to_string());
```

Comment:
Same here, leave it as a number ... it should be serialized properly into JSON number